ocaml/license/daily_license_check.ml

```diff
@@ -1,36 +1,47 @@
 module XenAPI = Client.Client
+module Date = Xapi_stdext_date.Date
 
 type result = Good | Expiring of string list | Expired of string list
 
-let seconds_per_day = 3600. *. 24.
+let a_month_after date =
+  let days_30 = Ptime.Span.unsafe_of_d_ps (30, 0L) in
+  Date.to_ptime date
+  |> (fun d -> Ptime.add_span d days_30)
+  |> Option.fold ~none:date ~some:Date.of_ptime
 
-let seconds_per_30_days = 30. *. seconds_per_day
+let days_to_expiry ~expiry now =
+  Ptime.diff (Date.to_ptime expiry) (Date.to_ptime now) |> Ptime.Span.to_d_ps
+  |> fun (d, ps) -> if ps = 0L then d else d + 1
```

Comment:
I don't understand this code. Can the diff be negative and do we have to take this into account?

Reply:
Hopefully now it's easier to understand the behaviour, and does the right thing